Crea los archivos necesarios para cumplir con la clase 6

ClassicalJo opened the pull request:
Intente mil veces subir todo esto, espero que esta vez salga bien.

ClassicalJo committed:
Crea los archivos necesarios para cumplir con la clase 6

Intente mil veces subir todo esto, espero que esta vez salga bien.

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
 Punto bonus: Crear un botón para "empezar de nuevo" que empiece el proceso nuevamente, borrando los inputs ya creados (investigar cómo en MDN).
 */
 
+//Crear inputs y labels
```
Este comentario no suma, si no entendés bien qué es `crearEntrada` ponele otro nombre que te lo explique mejor!

ClassicalJo replied:
Comentario borrado!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
 
+//Crear inputs y labels
+
+function crearEntrada(param) {
```
`param` no está claro qué hace, ponele un nombre más descriptivo. En este caso yo le pondría `indice` por ejemplo.

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    $labelSalario.className = "labelSalario"
+
+
+    nuevoParrafo = document.createElement("p")
```
siempre tenés que tener consistencia, si a todo lo demás le clavaste un `$`, este no tiene por qué ser la excepción!

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+
+
+
+generarArrays = function () {
```
a esto le falta una declaración, si no por default javascript te pone que es un `var`

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    }
+}
+
+//Recopilacion de datos del formulario en arrays
```
Este comentario tampoco suma, en vez de ponerle `generarArrays` le ponés `recopilarDatosUsuario` ponele y te ahorrás el comentario.

ClassicalJo replied:
comentario borrado y nombre de funcion cambiada!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    arrayInputEdad = arrayEnNumericos(arrayInputEdad)
+
+    $mostrarEdadMayor = document.querySelector("#mayorEdad")
+    $mostrarEdadMayor.value = calcularMayor(arrayInputEdad)
```
No escatimes con los nombres que son gratis. `calcularMayorEdad`

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+
+
+generarArrays = function () {
+    arrayInputEdad = document.querySelectorAll(".inputEdad")
```
2 cosas:

1. en general uno desacopla el tipo de elemento de la clase, es decir, le pone sólo `edad`.
2. en HTML y CSS, la nomenclatura es `kebab-case`, o sea que en este caso si decidieras mantener el nombre, sería `input-edad`.

ClassicalJo replied:
Decidi mantener la nomenclatura para no mezclarme yo porque me dio la impresion en su momento que si ponia edad como class se me iba a complicar a futuro en el proyecto e iba a tener que cambiar todo (lo mismo me paso con input-salario) pero definitivamente lo desaclopo en el proximo proyecto

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+        arrayInputSalarios = arrayEnNumericos(arrayInputSalarios)
+
+        $mostrarSalarioMayor = document.querySelector("#mayorSalario")
+        $mostrarSalarioMayor.value = calcularMayor(arrayInputSalarios) * 12
```
🤔 por qué lo mostrás anual? Y de ser así, prefiero que uses un nombre como el que pusiste abajo (`salario-anual-promedio`) antes que sólo `mayor-salario`.

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+
+}
+
+//convierte todos los elementos del array en valores numericos
```
`convertirArrayANumeros` y te ahorrás el comentario.

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+function arrayEnNumericos(array) {
+    let i;
+    let placeholder;
+    let dummyArray = [];
```
en vez de `dummy` se suele usar `temp` o `tmp` por "temporario".

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    return dummyArray
+}
+
+function calcularMayor(array) {
```
`function calcularMayorEdad(edades){ ... }`

ClassicalJo replied:
No hecho porque la funcion trabaja tambien en la determinacion del mayor salario. 
Si cambie el nombre por algo mas generico y claro.

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+
+
+
+clean = function () {
```
no uses Spanglish. En inglés o en español.

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    $salarioMensualPromedio = document.querySelector("#salarioMensualPromedio")
+    $salarioMensualPromedio.value = "Salario mensual promedio"
+    }
+    else {}
```
Podés borrar este `else{}` y va a funcionar igual, el `else` es opcional si precisás hacer algo en ese caso, si no, lo podés obviar.

Esto es válido:
```javascript
if(x === 1){
   //hacer algo sólo si x es igual 1
}
// no hace falta un else
```

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    $salarioAnualPromedio.value = "Salario anual promedio"
+
+    $salarioMensualPromedio = document.querySelector("#salarioMensualPromedio")
+    $salarioMensualPromedio.value = "Salario mensual promedio"
```
Todo esto tiene que estar identado adentro del `if`, te ayuda visualmente a entender cómo está estructurado el código

ClassicalJo replied:
hecho!

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+    else {}
+}
+
+clear = function () {
```
Creo `clean` y `clear` se prestan a confusión! Cuál es la diferencia entre cada uno?

ClassicalJo replied:
Clean borra el input del formulario y restaura los valores mostrados al inicio de los input deshabilitados.

Clear directamente empieza de nuevo el formulario (en la version corregida clear se llama empezarDeNuevo)

fsodano commented on tareas/tarea-6/tarea-clase-6.js:
```diff
@@ -6,11 +6,260 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
+
+
+
+
```
todas estas líneas están de más

ClassicalJo replied:
hecho!

ClassicalJo committed:
Cambia a kebab case todos los id

Renombre todos los ID;s en el HTML y en el .js acorde a kebab-case

ClassicalJo committed:
Cambia a Kebab-case los ID y correcciones

En resumen:
Cambie todos los IDS y clases en el codigo de camelCase a kebab-case. Es decir, todo lo que venga del HTML o que se cree en el HTML tiene kebab-case.

Borre todos los comentarios (que al fin y al cabo eran mas comentarios que solo tenian sentido para mi que para otras personas).

Aclare los nombres de las funciones para que sea mas claro lo que hacen, especialmente aquellas que tienen uso en mas de un array y no solo para un caso particular.

Reduje el spanglish, cambie placeholder y dummys por temp,

Y por ultimo borre las lineas vacias sin razon de ser.

ClassicalJo commented:
Corrije todo lo mencionado con anteroridad, con una excepcion! Gracias por el esfuerzo!

ClassicalJo committed:
aclara el nombre de funcion

Me olvide de agregar esto en el commit anterior. Cumple con la correccion sugerida!
